src/d/actor/d_a_player_main.cpp

```diff
@@ -2210,66 +2210,105 @@ int daPy_lk_c::getDirectionFromAngle(s16 angle) {
-    }
+void daPy_lk_c::setNormalSpeedF(f32 i_acceleration, f32 i_scale, f32 i_maxStep, f32 i_minStep) {
+   f32 desiredSpeed;
+   // Calculate the desired speed based on the stick distance and Link's maximum normal speed
+   if (dComIfGp_event_getMode() != dEvtMode_NONE_e || checkHeavyStateOn() || field_0x2b0 < 0.0f) {
+       desiredSpeed = mMaxNormalSpeed * mStickDistance;
+   }
+   else {
+       desiredSpeed = (mMaxNormalSpeed * mStickDistance) * mStickDistance;
+   }
+   if ((mDemo.getDemoMode() == daPy_demo_c::DEMO_UNK0E_e) || getSlidePolygon()) {
+       return;
+   }
+   // Special speed adjustment if Link collides with a wall
+   if (mAcch.ChkWallHit()) {
+       s16 angle = 0;
+       // Loop that determines which part of Link is touching the wall? (unsure about this)
+       dBgS_AcchCir* acchPtr = mAcchCir;
+       for (int i = 0; i < 3; i++, acchPtr++) {
+           if(acchPtr->ChkWallHit()) {
+               // Calculate the angle between Link and the wall
+               angle = current.angle.y + 0x8000 - acchPtr->GetWallAngleY();
+               break;
+           }
+       }
+       // If the angle is not too steep, reduce speed based on angle and slowdown factor
+       if (abs(angle) < 0x4000) {
+           desiredSpeed *= (1.0f - daPy_HIO_basic_c0::m.wallSlowdownFactor * cM_scos(angle));
+       }
+   }
+   f32 targetSpeed;
+   f32 maxStep;
+   // If slowing down (desired speed is less than current speed)
+   /* 
+       Note:
+       When `mVeloctiy`is negative, this conditional always evaluates to FALSE.
+   */
+   if (desiredSpeed < mVelocity) {
+       // Determine how much the Link should slow down by
+       f32 speedDelta = mVelocity - desiredSpeed;
+       // Limit how fast Link slows down
+       if (speedDelta > i_maxStep) {
+           maxStep = i_maxStep;
+       } else {
+           maxStep = speedDelta;
+       }
+       // Enforce a minimum deceleration step
+       if (maxStep < i_minStep) {
+           maxStep = i_minStep;
+       }
+       // External speed adjustments are ignored when slowing down
+       i_acceleration = 0.0f;
+       targetSpeed = desiredSpeed;
+   }
+   else {
+       // When accelerating, use the max step directly
+       maxStep = i_maxStep;
+       targetSpeed = 0.0f;
+   }
+   // Apply external speed additions if present
+   /*
+       Note: 
+       When this function is called from `setSpeedAndAngleSwim`, `i_acceleration` is the same as
+       `calculatedSpeed` and for any non-zero value `calculatedSpeed`, this conditional always evaluates to TRUE.
+   */
+   if (!cM3d_IsZero(i_acceleration)) {
+       /*
+           Note: 
+           Negative speed is added additively as well
+       */
+       mVelocity += i_acceleration;
+        // Clamp to the desired speed if it would exceed it
+        /*
+           Note:
+           This is purely a clamp on the positive speed and the lack of one 
+           for negative speed is what gives rise to super-swimming
+        */
```

Comment:
Since the purpose of this comment is to document a bug, it would be helpful to include the word "bug" in here somewhere so someone searching the codebase for bugs can find all of them relatively easily.